keras_cv/utils/fill_utils.py

```diff
@@ -10,42 +10,80 @@
+# limitations under the License.
 import tensorflow as tf
 
+from keras_cv.utils import bbox
 
-def fill_rectangle(
-    image, center_width, center_height, half_width, half_height, fill=None
-):
-    """Fill a rectangle in a given image using the value provided in replace.
+
+def rectangle_masks(mask_shape, corners):
+    """Computes positional masks of rectangles in images
 
     Args:
-        image: the starting image to fill the rectangle on.
-        center_width: the X center of the rectangle to fill
-        center_height: the Y center of the rectangle to fill
-        half_width: 1/2 the width of the resulting rectangle
-        half_height: 1/2 the height of the resulting rectangle
-        fill: A tensor with same shape as image. Values at rectangle
-         position are used as fill.
+        mask_shape: shape of the masks as [batch_size, height, width].
+        corners: rectangle coordinates in corners format.
+
     Returns:
-        image: the modified image with the chosen rectangle filled.
+        boolean masks with True at rectangle positions.
     """
-    image_shape = tf.shape(image)
-    image_height = image_shape[0]
-    image_width = image_shape[1]
-
-    lower_pad = tf.maximum(0, center_height - half_height)
-    upper_pad = tf.maximum(0, image_height - center_height - half_height)
-    left_pad = tf.maximum(0, center_width - half_width)
-    right_pad = tf.maximum(0, image_width - center_width - half_width)
-
-    shape = [
-        image_height - (lower_pad + upper_pad),
-        image_width - (left_pad + right_pad),
-    ]
-    padding_dims = [[lower_pad, upper_pad], [left_pad, right_pad]]
-    mask = tf.pad(tf.zeros(shape, dtype=image.dtype), padding_dims, constant_values=1)
-    mask = tf.expand_dims(mask, -1)
-
-    image = tf.where(tf.equal(mask, 0), fill, image)
-    return image
+    # add broadcasting axes
+    corners = corners[..., tf.newaxis, tf.newaxis]
+
+    # split coordinates
+    x0 = corners[:, 0]
+    y0 = corners[:, 1]
+    x1 = corners[:, 2]
+    y1 = corners[:, 3]
+
+    # repeat height and width
+    batch_size, height, width = mask_shape
+    x0_rep = tf.repeat(x0, height, axis=1)
+    y0_rep = tf.repeat(y0, width, axis=2)
+    x1_rep = tf.repeat(x1, height, axis=1)
+    y1_rep = tf.repeat(y1, width, axis=2)
+
+    # range grid
+    range_row = tf.range(0, height, dtype=corners.dtype)
+    range_col = tf.range(0, width, dtype=corners.dtype)
+    range_row = tf.repeat(range_row[tf.newaxis, :, tf.newaxis], batch_size, 0)
+    range_col = tf.repeat(range_col[tf.newaxis, tf.newaxis, :], batch_size, 0)
+
+    # boolean masks
+    mask_x0 = tf.less_equal(x0_rep, range_col)
+    mask_y0 = tf.less_equal(y0_rep, range_row)
+    mask_x1 = tf.less(range_col, x1_rep)
+    mask_y1 = tf.less(range_row, y1_rep)
+
+    masks = mask_x0 & mask_y0 & mask_x1 & mask_y1
+
+    return masks
+
+
+def fill_rectangle(images, center_x, center_y, width, height, fill):
```

Comment:
Isn't more explicit `centers.x` and `centers.y` `widths` and `heights` if we use `images` when we handle batches?

I think it Is quite a convention in TF API:
https://www.tensorflow.org/api_docs/python/tf/image/extract_patches